packages/affiliates/libs/affiliates.js

```diff
@@ -16,9 +16,11 @@ const DeployerRewards = ({ queries, empCreatorAbi, empAbi, coingecko, synthPrice
   assert(empAbi, "requires empAbi");
   assert(coingecko, "requires coingecko api");
   assert(synthPrices, "requires synthPrices api");
-  async function getBalanceHistory(address, start, end) {
+  async function getBalanceHistory(address) {
     // stream is a bit more optimal than waiting for entire query to return as array
-    const stream = await queries.streamLogsByContract(address, start, end);
+    // We need all logs from beginning of time. This could be optimized by deducing or supplying
+    // the specific emp start time to narrow down the query.
+    const stream = await queries.streamAllLogsByContract(address);
```

Comment:
This is the main logical change here, events to generate balances are queried for all time ( to get all contract events) vs previously used start and end time.

Reply:
Is there any reason why you can't manually set an end time, even if you must now default start time to deployment time?

Reply:
Could definitely be done that way, and I would consider that as an improvement/optimization to this. Really just wanted to get a fix in quickly without too much additional complexity though, because getting the start time of each emp was a bit ugly.

[made an issue for your suggestion](https://github.com/UMAprotocol/protocol/projects/6#card-49245181)

Reply:
Got it, yeah you can always add it later if you need it.